webhook/core/pod_webhook.go

```diff
@@ -76,14 +77,37 @@ func (prj *PodResourceInjector) Handle(ctx context.Context, req admission.Reques
 	//	pod.Spec.Containers[0].Resources.Limits[vpcresourceconfig.ResourceNameIPAddress] = resource.MustParse(resourceLimit)
 	//	pod.Spec.Containers[0].Resources.Requests[vpcresourceconfig.ResourceNameIPAddress] = resource.MustParse(resourceLimit)
 	//} else
+
+	fargatePodSgAnnotKey := "fargate.amazonaws.com/pod-sg"
+	isFargatePod := false
+	if _, ok := pod.ObjectMeta.Labels["eks.amazonaws.com/fargate-profile"]; ok {
+		isFargatePod = true
+	}
+	annotationMap := pod.ObjectMeta.Annotations
+	if annotationMap == nil {
+		annotationMap = make(map[string]string)
+	}
+
 	if sgList, cacheErr := prj.CacheHelper.GetPodSecurityGroups(pod); cacheErr != nil {
 		webhookLog.Error(cacheErr, "Webhook client failed to Get or List objects from cache.")
 		return admission.Denied("Webhood encountered error to Get or List object from k8s cache.")
-	} else if len(sgList) > 0 {
+	} else if len(sgList) > 0 && !isFargatePod {
 		webhookLog.Info("Injecting resource to the first container of the pod",
 			"resource name", vpcresourceconfig.ResourceNamePodENI, "resource count", resourceLimit)
 		pod.Spec.Containers[0].Resources.Limits[vpcresourceconfig.ResourceNamePodENI] = resource.MustParse(resourceLimit)
 		pod.Spec.Containers[0].Resources.Requests[vpcresourceconfig.ResourceNamePodENI] = resource.MustParse(resourceLimit)
+	} else if len(sgList) > 0 && isFargatePod {
+		// add pod-sg annotation if there is any matching security group in defined SGP (CRD)
+		annotationMap[fargatePodSgAnnotKey] = strings.Join(sgList, ",")
+		pod.ObjectMeta.Annotations = annotationMap
+		webhookLog.Info("Annotating fargate pod with security groups defined in CRD",
+			"Pod Security Group Annotation", annotationMap)
+	} else if len(sgList) == 0 && isFargatePod{
+		// override pod-sg annotation if it is not added outside of mutating webhook
```

Comment:
comment here is not correct, will be updated in the next revision. should be `// override pod-sg annotation if it is added before mutating webhook`

Reply:
nit: can you also add a bit of details on the reason behind it in the comment if possible. Something like the validating webhook will catch all the update on the given annotation key and the mutating webhook will catch create events outside of the webhook.